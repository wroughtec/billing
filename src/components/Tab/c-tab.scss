.c-tab {
  display: block;
  padding: .75rem .5rem;

  font-size: 1.25rem;

  color: var(--linkColour);

  text-align: center;
  text-decoration: none;

  &:hover {
    color: var(--linkHoverColour);
  }

  &.is-active {
    color: var(--linkActiveColour);

    background-color: var(--linkActiveBackgroundColour);
  }
}
