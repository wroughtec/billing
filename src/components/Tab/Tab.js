// @flow

import React, { Component } from 'react';
import { Link } from '@reach/router';
import './c-tab.scss';

type Props = {
  name: string,
  url: string
};

export class Tab extends Component<Props> {
  isActive = ({ isCurrent }: any) => (isCurrent ? { className: 'c-tab is-active' } : { className: 'c-tab' });

  render() {
    const { name, url } = this.props;

    return (
      <Link getProps={this.isActive} to={url}>
        {name}
      </Link>
    );
  }
}
