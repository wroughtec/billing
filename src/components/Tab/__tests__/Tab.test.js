import React from 'react';
import { shallow } from 'enzyme';
import { Tab } from '../Tab';

describe('Tab', () => {
  test('Renders', () => {
    const component = shallow(<Tab name="name" url="url" />);

    expect(component).toMatchSnapshot();
  });
});
