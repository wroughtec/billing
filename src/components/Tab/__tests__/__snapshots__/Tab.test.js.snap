// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tab
    name="name"
    url="url"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "name",
      "getProps": [Function],
      "to": "url",
    },
    "ref": null,
    "rendered": "name",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "name",
        "getProps": [Function],
        "to": "url",
      },
      "ref": null,
      "rendered": "name",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
